### ISSUE TITLE
[translation example]unable to test on new dataset after preprocessing

### ISSUE THREAD

**Issue by Tian312** [pytorch/fairseq#276]
I want to test the best model on new dataset that has not been preprocessed together with train/valid/test corpus before. 
when I tried to use preprocess.py to create a new dataset in data-bin and then use models from another dataset, there will be a size mismatch error from checkpoint for the model I applied. But if I try to replace the test corpus in original data-bin that I trained the best model on with the new test set, there will be a dictionary indexing issue. 

So far what I can do is to make sure every test examples are included when doing preprocessing and creating dictionaries before training. Is there any easy way to test the model on new dataset (e.g., 1 sentence) after the training is done?

(Or in short) how can I create data-bin using existing dictionaries?

**Comment by Tian312** [comment]
@edunov Thank you!
